src/allmydata/test/test_testing.py

```diff
@@ -0,0 +1,38 @@
+from twisted.trial.unittest import (
+    TestCase,
+)
+from twisted.internet.defer import (
+    inlineCallbacks,
+)
+
+from allmydata.testing.web import (
+    create_tahoe_treq_client,
+)
+
+
+class FakeWebTest(TestCase):
+    """
+    Test the WebUI verified-fakes infrastucture
+    """
+
+    def setUp(self):
+        super(FakeWebTest, self).setUp()
+        self.http_client = create_tahoe_treq_client()
+
+    @inlineCallbacks
+    def test_create_and_download(self):
+        """
+        Upload some content and then download it
+        """
+        content = "fake data\n" * 100
+        resp = yield self.http_client.put("http://example.com/uri", content)
+        cap = yield resp.content()
+
+        self.assertTrue(cap.startswith("URI:CHK:"))
```

Comment:
Would it make sense to parse `cap` with `allmydata.uri.from_string` here instead?  `URI:LIT` and `URI:CHK` both also carry a size so there could be an assertion that the size from the cap agrees with the real size of the data.

Reply:
The size is just 1000, 2000 etc as the caps are generated. Since the caps are "completely fake" anyway I don't see much value in plumbing through the file-size, personally.